libs/utils/decorate.js

```diff
@@ -68,5 +53,63 @@ export function decorateBlockBg(block, node) {
 export function getBlockSize(el, defaultSize = 1) {
   const sizes = ['small', 'medium', 'large', 'xlarge'];
   if (defaultSize < 0 || defaultSize > sizes.length - 1) return null;
-  return sizes.find((size) => el.classList.contains(size)) || sizes[defaultSize];
+  return sizes.find((size) => el?.classList.contains(size)) || sizes[defaultSize];
+}
+
+function getCopyDescendants(node, fragment = document.createDocumentFragment()) {
+  for (let i = 0; i < node.childNodes.length; i += 1) {
+    const child = node.childNodes[i];
+    fragment.appendChild(child.cloneNode(true));
+  }
+  return fragment;
+}
+
+export function decorateButtons(buttons) {
+  const mapBtnSize = { large: 'button-L', xlarge: 'button-XL' };
+  const isStrongOrEm = (node) => node.nodeName === 'STRONG' || node.nodeName === 'EM';
+  buttons.forEach((button) => {
+    const block = button.closest('.section div[class]:not(.content)');
+    const blockSize = getBlockSize(block);
+    const size = mapBtnSize[blockSize] ?? blockSize;
+    const parent = button.parentElement;
+
+    const child = button.childNodes?.length > 0
+      ? Array.from(button.childNodes).filter(isStrongOrEm)[0] : null;
+    const grandChild = child?.childNodes?.length > 0
+      ? Array.from(child.childNodes).filter(isStrongOrEm)[0] : null;
+    const nodes = [parent.nodeName, child?.nodeName, grandChild?.nodeName];
+    const text = parent.textContent || '';
+    const buttonTypes = [];
+    if (nodes.includes('STRONG') && nodes.includes('EM')) {
+      buttonTypes.push('fill');
+    } else if (nodes.includes('STRONG')) {
+      buttonTypes.push('blue');
+    } else if (nodes.includes('EM')) {
+      buttonTypes.push('outline');
+    }
+    button.classList.add('con-button', ...buttonTypes);
+    if (button.closest('.marquee')) {
+      // without this authors must review marquees in all projects to stop buttons having wrong size
+      button.classList.add(blockSize === 'large' ? 'button-XL' : 'button-L');
+    } else {
+      button.classList.add(size);
+    }
```

Comment:
Specific element styles for each block are not currently consistent across consonant blocks and instead of peppering conditional logic into these shared functions I think things like element size belongs in each respective block or in the blocks css styles as an override. In our shared text decorator we pass a config object to keep our variations sizes selected inside the block. This way we avoid nesting a bunch of conditional logic into our shared decorator functions. Maybe just something to think about as we add more consonant blocks and variations in the future.

Reply:
Agree in principle @Sartxi ! This is also happening for **all** blocks asides from the marquee. 
Let's take an aside for example. When you add an aside you can pass the size like so 
aside (large). 
If you do so, the buttons within will also be large. 
I'm not sure if you saw the code comment. I'm kind of forced to add this conditional logic for the marquee since I'm migrating code that was already written. The only way to not have this condition would be to go through every single marquee authored and adjust the size, which is obviously not something I want to force on the authors.

Reply:
I think this is Ok as is. But agree we'll probably want a refactor task to move these conditionals out and into their respective block decorators. Similar to the 'extendButtons' function in marquees.

Reply:
Can we not change our CSS to be more resilient to this stuff?

```
.marquee.xxl .con-button {
  height: 120px;
}
```

This would keep our JS simpler and keep the size overrides in the blocks that want to override.